src/drivers/indexeddb.js

```diff
@@ -648,28 +764,35 @@ function keys(callback) {
 
     var promise = new Promise(function(resolve, reject) {
         self.ready().then(function() {
-            var dbInfo = self._dbInfo;
-            var store = dbInfo.db.transaction(dbInfo.storeName, 'readonly')
-                .objectStore(dbInfo.storeName);
+            createTransaction(self._dbInfo, 'readonly', function(err, transaction) {
```

Comment:
I wonder if there is already a constant for `'readonly'` and `'readwrite'` we could use? if not, maybe we can make some rather than repeating the strings everywhere?